assignments/lambda-classes.js

```diff
@@ -1 +1,97 @@
 // CODE here for your Lambda Classes
+class Person {
+  constructor(personnel) {
+    this.name = personnel.name;
+    this.age = personnel.age;
+    this.location = personnel.location;
+    this.gender = personnel.gender;
+  }
+  speak() {
+    return `Hello my name is ${this.name}, I am from ${this.location}`;
+  }
+}
+
+class Instructor extends Person {
+  constructor(personnel) {
+    super(personnel);
+    this.speciality = personnel.speciality;
+    this.favLanguage = personnel.favLanguage;
+    this.catchPhrase = personnel.catchPhrase;
+  }
+  demo(subject) {
+    return `Today we are learning about ${subject}`;
+  }
+  grade(student) {
+    return `${student.name} receives a perfect score on ${student.subject}`;
+  }
+}
+
+class Student extends Person {
+  constructor(student) {
+    super(student);
+    this.previousBackground = student.previousBackground;
+    this.classname = student.classname;
+    this.favSubjects = student.favSubjects;
+  }
+  listsSubjects() {
+    let subjects = " ";
+    this.favSubjects.forEach(subject => {
```

Comment:
You get undefined when you log the function to the console.
You could want to refactor this function to return a string with all the values of your array.

Looks like you started the process here with that subjects string but didn't finished it.